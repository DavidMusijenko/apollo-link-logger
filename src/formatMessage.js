const formatMessage = (operationType, operation, ellapsed) => {
  const headerCss = [
    'color: gray; font-weight: lighter', // title
    `color: ${operationType === 'query' ? '#03A9F4' : 'red'};`, // operationType
    'color: inherit;', // operationName
    'color: gray; font-weight: lighter;', // time, etc
  ];

  const parts = ['%c apollo'];

  parts.push(`%c${operationType}`);
  parts.push(`%c${operation.operationName}`);

  if (operationType !== 'subscription') {
    parts.push(`%c(in ${ellapsed} ms)`);
  }

  return [parts.join(' '), ...headerCss];
};

export default formatMessage;
