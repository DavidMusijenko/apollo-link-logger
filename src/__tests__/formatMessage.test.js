import formatMessage from '../formatMessage';

describe('formatMessage', () => {
  it('returns a list of strings suitable for console logging', () => {
    expect(
      formatMessage('query', { operationName: 'FormatMessageQuery' }, 1000),
    ).toMatchSnapshot();

    expect(
      formatMessage(
        'mutation',
        { operationName: 'FormatMessageMutation' },
        500,
      ),
    ).toMatchSnapshot();
  });
});
