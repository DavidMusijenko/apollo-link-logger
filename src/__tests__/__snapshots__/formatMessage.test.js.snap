// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatMessage returns a list of strings suitable for console logging 1`] = `
Array [
  "%c apollo %cquery %cFormatMessageQuery %c(in 1000 ms)",
]
`;

exports[`formatMessage returns a list of strings suitable for console logging 2`] = `
Array [
  "%c apollo %cmutation %cFormatMessageMutation %c(in 500 ms)",
]
`;
